.changes/unreleased/FEATURES-20240404-155023.yaml

```diff
@@ -0,0 +1,8 @@
+kind: FEATURES
+body: 'types/validation: Added `BoolValuableWithValidateableParameter`, `Float64ValuableWithValidateableParameter`,
```

Comment:
The utility of the value-type interfaces is debatable as the interface that they embed, `ValidateableParameter`, can be implemented directly by a custom value type. However, the value-type interfaces do streamline defining the interfaces that a custom value type should implement. For example

Using `validation.ValidateableParameter`:

```go
var _ basetypes.StringValuable = CustomStringValue{}
var _ validation.ValidateableParameter = CustomStringValue{}
```

Using `validation. StringValuableWithValidateableParameter`:
```go
var _ validation.StringValuableWithValidateableParameter = CustomStringValue
```